lib/trackler/doc_file.rb

```diff
@@ -0,0 +1,92 @@
+class DocFile
+  DEFAULT_IMAGE_PATH = "/docs/img"
+
+  def self.find(basename:, track_dir:, image_path:)
+    dir = File.join(track_dir, "docs")
+
+    [
+      MarkdownFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      OrgmodeFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      NullFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+    ].detect(&:acceptable?)
+  end
+
+  attr_reader :basename, :dir, :image_path
+  def initialize(basename:, docs_dir:, image_path:)
+    @basename = basename
+    @dir = docs_dir
+    @image_path = image_path
+  end
+
+  def to_md
+    transform(body).gsub(img_src, img_dst)
```

Comment:
Something about the image path substitution here is bugging me. 
I suspect it is a separate operation from converting to markdown.
(See comment about abstractions below)